src/ngx-smart-loader/src/services/ngx-smart-loader.service.ts

```diff
@@ -184,4 +183,51 @@ export class NgxSmartLoaderService {
     }
   }
 
+  /**
+   * Get the loader from the loaderStack and start it
+   * If the loader is NOT created yet, we store the start action as a promise to execute it at the loader creation
+   *
+   * @param identifier The loader identifier.
+   * */
+  private doStart(identifier): void {
+    const loader = this.getLoader(identifier);
+    // The loader is not created yet, I store the start action to execute it when the loader will be added
+    if (!loader) {
+      this.addWaitingAction(identifier, 'onStart', this.doStart);
+      return;
+    }
+
+    loader.start();
+  }
+
+  /**
+   * Get the loader from the loaderStack and stop it
+   * If the loader is NOT created yet, we store the stop action as a promise to execute it at the loader creation
+   *
+   * @param identifier The loader identifier.
+   * */
+  private doStop(identifier): void {
+    const loader = this.getLoader(identifier);
+    // The loader is not yet created, I store the stop action to execute it when the loader will be added
+    if (!loader) {
+      this.addWaitingAction(identifier, 'onStop', this.doStop);
+      return;
+    }
+
+    loader.stop();
+  }
+
+  private addWaitingAction(identifier: string, action: string, callback: (id: string) => void) {
+    const actions = this.waitingActions.get(identifier) || {};
+    actions[action] = callback.bind(this);
+    this.waitingActions.set(identifier, actions);
+  }
+
+  private executeWaitingAction(identifier: string, action: string): void {
+    const actions = this.waitingActions.get(identifier);
+    if (actions && actions.hasOwnProperty(action)) {
+      actions[action]();
+      delete actions[action];
```

Comment:
actions is an instance of ActionsToExecute, questions: delete a key will make it invalid how is that possible. Also is there a real need to delete this action ? Shouldn't it be execute only once? Otherwise it's a problem because it should not come here many time, don't you think?

Reply:
The previous line to the delete executes the callback function that has been registered. 
After executes it I remove it. We are in a service, the start method can be executed multiple time on a same loader

Reply:
wouldn't it be normal to execute same callbacks on each call of the start ?